src/views/Player.vue

```diff
@@ -267,28 +263,46 @@ export default defineComponent({
       if (profile.value && battleTag.value === profile.value.battleTag) return;
 
       await playerStore.loadProfile({ battleTag: battleTag.value, freshLogin: props.freshLogin });
-      await playerStore.loadGameModeStats({});
-      await playerStore.loadRaceStats();
-      await playerStore.loadPlayerStatsRaceVersusRaceOnMap(battleTag.value);
-      await playerStore.loadPlayerStatsHeroVersusRaceOnMap(battleTag.value);
-      await playerStore.loadPlayerGameLengths();
-      await playerStore.loadMatches(1);
+      await loadAllDataForSelectedSeason();
       await initMmrRpTimeline();
       window.scrollTo(0, 0);
     }
 
+    async function loadAllDataForSelectedSeason() {
+      await Promise.all([
+        playerStore.loadGameModeStats({}),
+        playerStore.loadRaceStats(),
+        playerStore.loadMatches(1),
+        playerStore.loadPlayerStatsRaceVersusRaceOnMap(battleTag.value),
+        playerStore.loadPlayerStatsHeroVersusRaceOnMap(battleTag.value),
+        playerStore.loadPlayerGameLengths(),
+      ]);
+    }
+
     async function initMmrRpTimeline() {
-      const raceStats = playerStore.raceStats;
+      const selectedSeason = playerStore.selectedSeason?.id;
+      let maxMode = EGameMode.GM_1ON1;
+      let maxModeGames = 0;
+      playerStore.gameModeStats.forEach((m) => {
+        if (m.season !== selectedSeason) return;
```

Comment:
Does this guard protect against some specific scenario where the gameModeStats and selected season aren't the same?

Reply:
Good question, I'll retest without this, I remember this fixing the result when I wrote it earlier but it may no longer be necessary after the changes to the await loading

Reply:
You're right, wasn't needed anymore. But I did realize when clicking on old seasons, some players played some inactive modes the most, so I added a check to skip inactive modes when picking a favourite so it tends to pick one we can show an MMR graph for